src/overlays/actors/ovl_Obj_Mu_Pict/z_obj_mu_pict.c

```diff
@@ -21,30 +30,164 @@ const ActorInit Obj_Mu_Pict_InitVars = {
+    
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if (this->actor.cutscene < 0) {
+            func_80C06DC8(this, globalCtx);
+            func_80C06CC4(this);
+        } else {
+            if (ActorCutscene_GetCurrentIndex() == 0x7c) {
+                ActorCutscene_Stop(0x7c);
+            }
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+            func_80C06DC8(this, globalCtx);
+            func_80C06C54(this);
+        }
+    } else if (ABS_ALT(yawDiff) < 0x2AAA) {
+        func_800B8614(&this->actor, globalCtx, 80.0f);
+    }
+}
+
+void func_80C06C54(ObjMuPict* this) {
+    this->actionFunc = func_80C06C68;
+}
+
+void func_80C06C68(ObjMuPict* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->actor.cutscene) != 0) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        func_80C06CC4(this);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
+
+void func_80C06CC4(ObjMuPict* this) {
+    this->actionFunc = func_80C06CD8;
+}
+
+void func_80C06CD8(ObjMuPict* this, GlobalContext* globalCtx) {
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 0:
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+            break;
+        case 5:
+            func_80C06E88(this, globalCtx);
+            break;
+        case 6:
+            if (func_80147624(globalCtx)) {
+                func_80C06B5C(this);
+                if (this->actor.cutscene >= 0) {
+                    ActorCutscene_Stop(this->actor.cutscene);
+                }
+            }
+            break;
+    }
+    Math_SmoothStepToS(&this->actor.parent->world.rot.y, this->actor.parent->yawTowardsPlayer, 0xA, 0x71C, 0xB6);
+    this->actor.parent->shape.rot.y = this->actor.parent->world.rot.y;
+}
+
+void func_80C06D90(ObjMuPict* this, GlobalContext* globalCtx) {
+    struct Actor* actorPtr = globalCtx->actorCtx.actorList[ACTORCAT_NPC].first;
+
+    while (actorPtr != NULL) {
+        if (actorPtr->id == ACTOR_EN_HGO) {
+            this->actor.parent = actorPtr;
+            return;
+        }
+        actorPtr = actorPtr->next;
+    }
+}
+
+void func_80C06DC8(ObjMuPict* this, GlobalContext* globalCtx) {
+    if (this->unk14A == 0) {
+        if (this->unk148 == 0) {
+            this->unk148 = 1;
+            func_801518B0(globalCtx, 0x159A, &this->actor);
+            this->textId = 0x159A;
+        } else {
+            func_801518B0(globalCtx, 0x159D, &this->actor);
+            this->textId = 0x159D;
+        }
+    } else if (this->unk148 == 0) {
+        this->unk148 = 1;
+        func_801518B0(globalCtx, 0x15A0, &this->actor);
+        this->textId = 0x15A0;
+    } else {
+
+        func_801518B0(globalCtx, 0x15A3, &this->actor);
+        this->textId = 0x15A3;
+    }
+}
+
+void func_80C06E88(ObjMuPict* this, GlobalContext* globalCtx) {
+    if (func_80147624(globalCtx)) {
+        switch (this->textId + 1) {
```

Comment:
```suggestion
        switch (this->textId) {
```
This is tricky. Should you include the `+1`. I see how the cases line up nicely if you include the `+1` in the switch instead of shifting all the other cases up by one. Thoughts on this?

Reply:
I say keep the `+1` out of it. This looks to be a the text handler in a conversation, i.e., case 0x159A would occur when that textbox appears, which then sets 0x159B or the next textbox in the conversation.